frontend/src/lib/components/SelectBox.tsx

```diff
@@ -0,0 +1,248 @@
+                    return (
+                        props.items.filter((i) => i.dataSource.filter((i) => i.key === item.key).length > 0)[0]
+                            ?.renderInfo || false
+                    )
+                },
+            },
+        ],
+    }),
+    listeners: ({ props, values, actions }) => ({
+        clickSelectedItem: ({ item }: { item: SelectedItem }) => {
+            if (item.event) {
+                props.updateFilter(EntityTypes.EVENTS, item.event, item.event)
+            } else {
+                props.updateFilter(EntityTypes.ACTIONS, item.action.id, item.action.name)
+            }
+        },
+        setBlockMouseOver: ({ block }) => {
+            if (block) setTimeout(() => actions.setBlockMouseOver(false), 200)
+        },
+        onKeyDown: ({ e }) => {
+            let allSources = props.items.map((item) => item.dataSource).flat()
+            allSources = !values.search
+                ? allSources
+                : new Fuse(allSources, {
+                      keys: ['name'],
+                  })
+                      .search(values.search)
+                      .map((result) => result.item)
+            const currentIndex = allSources.findIndex((item: SelectedItem) => item.key === values.selectedItem.key) || 0
+
+            if (e.key === 'ArrowDown') {
+                const item = allSources[currentIndex + 1]
+                if (item) {
+                    actions.setSelectedItem(item)
+                    scrollDownIntoView(item.key)
+                    actions.setBlockMouseOver(true)
+                }
+            }
+            if (e.key === 'ArrowUp') {
+                const item = allSources[currentIndex - 1]
+                if (item) {
+                    actions.setSelectedItem(item)
+                    scrollUpIntoView(item.key)
+                    actions.setBlockMouseOver(true)
+                }
+            }
+            if (e.key === 'Enter') {
+                actions.clickSelectedItem(values.selectedItem)
+            }
+        },
+    }),
+})
+
+export function SelectBox({
+    items,
+    selectedItemKey,
+    onSelect,
+    onDismiss,
+}: {
+    items: SelectBoxItem[]
+    selectedItemKey: string
+    onSelect: CallableFunction
+    onDismiss: CallableFunction
+}): JSX.Element {
+    const dropdownRef = useRef()
+    const dropdownLogic = actionFilterDropdownLogic({ updateFilter: onSelect, items })
+    const { selectedItem, RenderInfo } = useValues(dropdownLogic)
+    const { setSearch, setSelectedItem, onKeyDown } = useActions(dropdownLogic)
+
+    const deselect = (e): void => {
+        if (dropdownRef.current.contains(e.target)) {
+            return
+        }
+        onDismiss && onDismiss(e)
+    }
+
+    useEffect(() => {
+        if (selectedItemKey) {
+            const allSources = items.map((item) => item.dataSource).flat()
+            setSelectedItem(allSources.filter((item) => item.key === selectedItemKey)[0] || false)
```

Comment:
why is the || false needed here?

Reply:
I'm actually not sure it is. I ran into this while developing but can't reproduce it anymore. I do still think it's safer, as the alternative is having a complete white-out